scipy/optimize/tests/test__sda.py

```diff
@@ -0,0 +1,120 @@
+# Simulated dual annealing unit tests implementation.
+# Copyright (c) 2017 Sylvain Gubian <sylvain.gubian@pmi.com>,
+# Yang Xiang <yang.xiang@pmi.com>
+# Author: Sylvain Gubian, PMP S.A.
+##############################################################################
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+##############################################################################
+"""
+Unit tests for the simulted dual annealing global optimizer
+"""
+from .._sda import sda
+from .._sda import VisitingDistribution
+from .._sda import ObjectiveFunWrapper
+from .._sda import EnergyState
+import numpy as np
+from numpy.testing import (assert_equal, TestCase, assert_allclose,
+                           assert_almost_equal, assert_raises,
+                           assert_array_less)
+from scipy._lib._util import check_random_state
+
+
+class TestSDA(TestCase):
+
+    def setUp(self):
+        # Using Rastrigin function for performing tests
+        self.func = lambda x: np.sum(x * x - 10 * np.cos(
+            2 * np.pi * x)) + 10 * np.size(x)
+        # A function that returns always a big value for initialization tests
+        self.weirdfunc = lambda x: 2e16
+        self.ld_bounds = [(-5.12, 5.12)] * 2
+        self.hd_bounds = self.ld_bounds * 4
+        self.nbtestvalues = 5000
+        self.high_temperature = 5230
+        self.low_temperature = 0.1
+        self.qv = 2.62
+        self.defautgr = (self.func, None, self.ld_bounds)
+        self.seed = 1234
+        self.rs = check_random_state(self.seed)
+
+    def tearDown(self):
+        pass
+
+    def test_low_dim(self):
+        ret = sda(self.func, None, self.ld_bounds, seed=self.seed)
+        assert_allclose(ret.fun, 0., atol=1e-12)
+
+    def test__visiting_stepping(self):
+        lu = list(zip(*self.ld_bounds))
+        lower = np.array(lu[0])
+        upper = np.array(lu[1])
+        dim = lower.size
+        vd = VisitingDistribution(lower, upper, self.qv, self.rs)
+        values = np.zeros(dim)
+        x_step_low = vd.visiting(values, 0, self.high_temperature)
+        # Make sure that only the first component is changed
+        assert np.all(x_step_low != 0)
```

Comment:
No plain asserts please (those get stripped by `python -O`). Use `np.testing.assert_` instead.

Reply:
Changed to numpy asserts.